Plugins/Public/minecontrol_plugin/Main.cpp

```diff
@@ -352,132 +271,47 @@ EXPORT void LoadSettings()
-							{
-								Archetype::Gun* gun = (Archetype::Gun*) eq;
-								if (gun->iProjectileArchID && gun->iProjectileArchID != 0xBAADF00D && gun->iProjectileArchID != 0x3E07E70)
-								{
-									pb.lstAmmo.push_back(gun->iProjectileArchID);
-								}
-							}
-							else
-							{
-								pb.lstItems.push_back(iItemID);
-							}
-						}
-						else if (Archetype::GetSimple(iItemID))
-						{
-							pb.lstItems.push_back(iItemID);
-						}
-						else
-						{
-							ConPrint(L"ERROR: %s:%d: item '%s' not valid\n", stows(ini.get_file_name()).c_str(), ini.get_line_num(), stows(scShipOrEquip).c_str());
-						}
-
-						scShipOrEquip = GetTrimParam(scLine, i++);
-					}
-
-					set_mmapPlayerBonus.insert(multimap<uint, PLAYER_BONUS>::value_type(pb.iLootID, pb));
-					if (set_iPluginDebug)
-					{
-						ConPrint(L"NOTICE: mining player bonus %s(%u) %2.2f %s(%u)\n",
-							stows(GetTrimParam(scLine, 0)).c_str(), pb.iLootID, pb.fBonus, stows(GetTrimParam(scLine, 2)).c_str(), pb.iRep);
+						idBonusMap[licenceId][lootId] = bonus;
 					}
 				}
 			}
 			else if (ini.is_header("ZoneBonus"))
 			{
 				while (ini.read_value())
 				{
-					string scLine = ini.get_name_ptr();
-
-					string scZone = GetTrimParam(scLine, 0);
-					if (!scZone.size())
+					if (ini.is_value("zb"))
 					{
-						ConPrint(L"ERROR: %s:%d: zone not valid\n", stows(ini.get_file_name()).c_str(), ini.get_line_num());
-						continue;
-					}
-
-					float fBonus = (float)atof(GetTrimParam(scLine, 1).c_str());
-					if (fBonus <= 0.0f)
-					{
-						ConPrint(L"ERROR: %s:%d: bonus not valid\n", stows(ini.get_file_name()).c_str(), ini.get_line_num());
-						continue;
-					}
-
-					uint iReplacementLootID = 0;
-					string scReplacementLoot = GetTrimParam(scLine, 2);
-					if (scReplacementLoot.size())
-					{
-						iReplacementLootID = CreateID(scReplacementLoot.c_str());
-					}
-
-					//
-					float fRechargeRate = 1000;
-					//float fRechargeRate = (float)atof(GetTrimParam(scLine, 3).c_str());
-					//if (fRechargeRate <= 0.0f)
-					//{
-					//	fRechargeRate = 50000;					
-					//}
-
-					float fMaxReserve = 100000;
-					//float fMaxReserve = (float)atof(GetTrimParam(scLine, 4).c_str());
-					//if (fMaxReserve <= 0.0f)
-					//{
-					//	fMaxReserve = 100000;					
-					//}
-
-					uint iZoneID = CreateID(scZone.c_str());
-					set_mapZoneBonus[iZoneID].scZone = scZone;
-					set_mapZoneBonus[iZoneID].fBonus = fBonus;
-					set_mapZoneBonus[iZoneID].iReplacementLootID = iReplacementLootID;
-					set_mapZoneBonus[iZoneID].fRechargeRate = fRechargeRate;
-					set_mapZoneBonus[iZoneID].fMaxReserve = fMaxReserve;
-					if (set_iPluginDebug)
-					{
-						ConPrint(L"NOTICE: zone bonus %s fBonus=%2.2f iReplacementLootID=%s(%u) fRechargeRate=%0.0f fMaxReserve=%0.0f\n",
-							stows(scZone).c_str(), fBonus, stows(scReplacementLoot).c_str(), iReplacementLootID, fRechargeRate, fMaxReserve);
+						uint zoneID = CreateID(ini.get_value_string(0));
```

Comment:
Are you handling if there are missing parameters? ConLog stuff that is wrong or missing